src/Microsoft.AspNetCore.Http/HttpContextFactory.cs

```diff
@@ -1,30 +1,59 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
+using System.Text;
 using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Http.Features.Internal;
+using Microsoft.Extensions.ObjectPool;
 
 namespace Microsoft.AspNetCore.Http.Internal
 {
     public class HttpContextFactory : IHttpContextFactory
     {
-        private IHttpContextAccessor _httpContextAccessor;
+        private readonly ObjectPool<StringBuilder> _builderPool;
+        private readonly IHttpContextAccessor _httpContextAccessor;
 
-        public HttpContextFactory() : this(httpContextAccessor: null)
+        public HttpContextFactory()
+            : this(poolProvider: null, httpContextAccessor: null)
+        {
+        }
+
+        public HttpContextFactory(ObjectPoolProvider poolProvider)
+            : this(poolProvider, httpContextAccessor: null)
         {
         }
 
         public HttpContextFactory(IHttpContextAccessor httpContextAccessor)
+            : this(poolProvider: null, httpContextAccessor: httpContextAccessor)
         {
+        }
+
+        public HttpContextFactory(ObjectPoolProvider poolProvider, IHttpContextAccessor httpContextAccessor)
```

Comment:
Not testing constructor overloads w/ a non-`null` provider b/c that requires something ugly like passing a `Mock<IFeatureCollection>` to `Create()` (test project does not currently use `Moq`) or exposing an otherwise-useless property on the `ResponseCookiesFeature`.